File: python/laba1.py
import math

def solve(a, b, c):
    D = b ** 2 - 4 * a * c
    if(D < 0): #check frist sqrt
        return None
    x1 = (-b + math.sqrt(D)) / (2 * a)
    x2 = (-b - math.sqrt(D)) / (2 * a)
    
    roots = []
    
    if x1 >= 0: #roots of биквадратное
        roots.append(math.sqrt(x1))
        roots.append(-math.sqrt(x1))
    
    if x2 >= 0:
        roots.append(math.sqrt(x2))
        roots.append(-math.sqrt(x2))
    
    if len(roots) == 0:
        return None  
    
    return roots

def main():
    while(True):
        try:
            coef = list(map(float, input().split())) #разбиение ввода
            break
        except ValueError:
            print("Error")
    a = coef[0] #coefs
    b = coef[1]
    c = coef[2]
    result = solve(a, b, c)
    if result is None:
        print(None)
    else:
        print(result)

main()

File: sharp/laba1.cs
using System;
using System.Collections.Generic;

class Program
{
    static List<double> Solve(double a, double b, double c)
    {
        double D = b * b - 4 * a * c;
        if (D < 0)
        {
            return null; 
        }

        double x1 = (-b + Math.Sqrt(D)) / (2 * a);
        double x2 = (-b - Math.Sqrt(D)) / (2 * a);

        HashSet<double> roots = new HashSet<double>(); //unic roots

        // Корни биквадратного уравнения
        if (x1 >= 0)
        {
            roots.Add(Math.Sqrt(x1));
            roots.Add(-Math.Sqrt(x1));
        }

        if (x2 >= 0)
        {
            roots.Add(Math.Sqrt(x2));
            roots.Add(-Math.Sqrt(x2));
        }

        return roots.Count > 0 ? new List<double>(roots) : null; // тернарный оператор)) если кол-во,то массив иначе none
    }

    static void Main()
    {
        double a, b, c;
        System.Globalization.CultureInfo.CurrentCulture = System.Globalization.CultureInfo.InvariantCulture; // чтобы точки хавал
        
        while (true)
        {
            string input = Console.ReadLine();
            string[] parts = input.Split();

            if (parts.Length == 3 && //check counts of coef
                double.TryParse(parts[0], out a) && //перевод в double
                double.TryParse(parts[1], out b) &&
                double.TryParse(parts[2], out c))
            {
                break; 
            }
            else
            {
                Console.WriteLine("error"); 
            }
        }

        List<double> result = Solve(a, b, c);
        
        if (result == null)
        {
            Console.WriteLine("Действительных корней нет");
        }
        else
        {
            Console.WriteLine(string.Join(" ", result)); 
        }
    }
}
